_episodes/script.md

```diff
@@ -1,38 +1,115 @@
 ---
+
 By the end of this episode,
 learners should be able to
-__include their own script as a step in a workflow__
-
-several options for this were identified during the lesson development sprint.
-we should choose one or two to focus on in the tutorial
-(add your thoughts at the corresponding issue [here](https://github.com/common-workflow-lab/cwl-novice-tutorial/issues/8)):
-  - make script executable and add to path
-      - advantage: quick
-      - downside:
-          - the scripts are not shipped with the CWL document itself
-          - not portable to other execution environment without putting the script first
-  - pack the scripts into a docker container and make them executable there:
-      - advantage: portable
-      - downside:
-          - only works for people that are using containers
-  - use InitialWorkDirRequirement to list script content directly in the CWLToolWrapper
-      - example: https://github.com/CompEpigen/ATACseq_workflows/blob/master/CWL/tools/generate_atac_signal_tags.cwl (bad example as here the scripts are too long, only for short scripts)
-      - advantage:
-          - portable
-          - can be used independent of the dependency management solution
-      - disadvantage:
-          - explodes the complexity of a CWL tool wrapper
-          - having to deal with escapes and so on
-  - distribute as seperate package via pip / cran / ...
+__include and run their own script in a step at runtime__
+
+> ## Exercise 1:
+>
+> Identify `requirements` for the `CommandLineTool`. HINT: `'ubuntu:latest'` can be used as docker container.
+>
+> > ## Solution
+> > 
+> > requirements:
+> >    DockerRequirement:
+> >      dockerPull: 'ubuntu:latest'
+> >    InlineJavascriptRequirement: {}
+> >    InitialWorkDirRequirement:
+> > 
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 2:
+> 
+> Create an input field (with appropriate `type`) that is going to be referred within the script.
+>
+> > ## Solution:
+> >
+> > inputs:
+> >   message:
+> >     type: string
+> > 
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 3:
+>
+> Define name and content of the script (in this exercise we will create a bash script that uses `echo` on the input `message`)
+>
+> > ## Solution:
+> >
+> > InitialWorkDirRequirement:
+> >   listing:
+> >     - entryname: script.sh
+> >       entry: |
+> >         echo "*Documenting input*" && \
+> >         echo "Input received: $(inputs.message)" && \
+> >         echo "Exit"
+> > 
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 5:
+> 
+> Since we are using `echo` in the script - what is the apprproiate `CWL` feature to capture standard output?
+>
+> > ## Solution:
+> >
+> > stdout: "message.txt"
+> > 
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 6:
+> 
+> Create an output variable to capture this tools' output and identify an appropriate `type` for it.
+>
+> > ## Solution:
+> >
+> > outputs:
+> >   doc:
+> >     type: stdout
+> >
+> {: .solution}
+{: .challenge}
```

Comment:
I suspect this one would also be difficult for a novice. A fill-in-the-blanks, or some other format that provides some structure for the learner to start from, would help guide them to focus on the concept of capturing output and documenting its type.

Reply:
Updated this exercise to use fill in the blank type question [36e137d](https://github.com/common-workflow-lab/cwl-novice-tutorial/pull/20/commits/36e137d8102d6bad6b8cbf3a3d82ac39dd0c03e2)